### PR TITLE
[1LP][RFR] Enforce Good Collections

### PR DESCRIPTION
**Pull request by psav**
A proof of concept to enforce good collection objects

### CODE REVIEW

File: cfme/utils/appliance/__init__.py [hunk 1]
```diff
@@ -2625,8 +2626,24 @@ def __init__(self, appliance=None):
 
 
 class BaseCollection(NavigatableMixin):
-    pass
+    def __new__(cls, *args, **kwargs):
+        if inspect.getargspec(cls.__init__).args[1] != 'appliance':
```
Comment (izapolsk) [inline cfme/utils/appliance/__init__.py]: what if appliance isn't passed at all?

Comment (mshriver) [reply cfme/utils/appliance/__init__.py]: This is the IndexError that I saw that only comes from especially 'dumb' implementation/use

File: cfme/utils/appliance/__init__.py [hunk 2]
```diff
@@ -2668,8 +2727,40 @@ def __init__(self, appliance=None):
 
 
 class BaseCollection(NavigatableMixin):
-    pass
+    """Class for helping create consistent Collections
+
+    The BaseCollection class is responsible for ensuring two things:
+
+    1) That the API consistently has the first argument passed to it
+    2) That that first argument is an appliance
```
Comment (mshriver) [inline cfme/utils/appliance/__init__.py]: '...argument is an appliance instance'

File: cfme/utils/appliance/__init__.py [hunk 3]
```diff
@@ -2668,8 +2727,40 @@ def __init__(self, appliance=None):
...
 
 
 class BaseEntity(NavigatableMixin):
-    pass
+    """Class for helping create consistent entitys
+
+    The BaseEntity class is responsible for ensuring two things:
+
+    1) That the API consistently has the first argument passed to it
+    2) That that first argument is a collection
```
Comment (mshriver) [inline cfme/utils/appliance/__init__.py]: '...argument is a collection instance'

**Pull request merged**